rest/route/github.go

```diff
@@ -451,10 +450,33 @@ func (gh *githubHookApi) handleGitTag(ctx context.Context, event *github.PushEve
 
 func (gh *githubHookApi) createVersionForTag(ctx context.Context, pRef model.ProjectRef, existingVersion *model.Version,
 	revision model.Revision, tag model.GitTag) (*model.Version, error) {
-	if !pRef.GitTagVersionsEnabled || !utility.StringSliceContains(pRef.GitTagAuthorizedUsers, tag.Pusher) {
+	token, err := gh.settings.GetGithubOauthToken()
+	if err != nil {
+		return nil, errors.New("error getting github token")
+	}
+	if !pRef.GitTagVersionsEnabled {
 		return nil, nil
 	}
-	hasAliases, remotePath, err := gh.sc.HasMatchingGitTagAliasAndRemotePath(pRef.String(), tag.Tag)
+	authorized, err := pRef.AuthorizedForGitTag(ctx, tag.Pusher, token)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error determining if user '%s' is authorized", tag.Pusher)
+	}
+	if !authorized {
+		grip.Debug(message.Fields{
+			"source":           "github hook",
+			"msg_id":           gh.msgID,
+			"event":            gh.eventType,
+			"project":          pRef.Identifier,
+			"tag":              tag,
+			"message":          "user not authorized for git tag version",
+			"authorized_users": pRef.GitTagAuthorizedUsers, // TODO: remove these
```

Comment:
is this todo still needed?

Reply:
ah yeah i need  to finish testing with the new keys but i'll remove before merging